### ISSUE TITLE
Support for template strings

### ISSUE THREAD

**Issue by scottohara** [leafgarland/typescript-vim#54]
Typescript 1.4 introduced support for ES6-style template strings, eg.

```
var myString = `Hello, ${name}`;
```

Currently any tokens in the templated string (i.e. `${name}`) appear the same color as the literal parts.  Here's an example:

![image](https://cloud.githubusercontent.com/assets/289327/8048364/ad64f272-0e93-11e5-8464-fe43464a8a4f.png)

By contrast, [another plugin](https://github.com/othree/yajs.vim) I use for JS/ES6 highlighting differentiates between the tokens and the literal, eg:

![image](https://cloud.githubusercontent.com/assets/289327/8048356/96752e9c-0e93-11e5-8c53-6882b15be669.png)

Clearly it's a matter of opinion as to which looks better; I'm raising this issue simply because I wasn't sure if it was implemented on purpose like this.

**Comment by leafgarland** [comment]
Yes, it would be good to support that syntax highlighting - I had a go at it a while back but encountered some issues with folding. Haven't had chance to investigate further.

**Comment by jez** [comment]
For what it's worth, CoffeeScript [has similar functionality][1]. Could be useful as a jumping-off point.

[1]: <https://github.com/kchmck/vim-coffee-script/blob/master/syntax/coffee.vim#L139-L141>
